### DISCUSSION TITLE
header rowspan dan colspan feature

### DISCUSSION THREAD

**Discussion by ajikur2901** [rappasoft/laravel-livewire-tables#1122] [Q&A]
Hi, how i could add colpsan or rowspan to the table header like image below ?

![image](https://user-images.githubusercontent.com/48973903/227828015-53143eaf-acb9-44f4-aee5-c0b4ba206ac4.png)

Thanks!

**Comment by lrljoe** [comment]
You could use the secondary header method, and set the th to colspan using the attributes method in configure().

If that makes no sense then please reply so that I can clarify

If you think that Column groups as a whole would be a useful feature, then please raise a feature request so that the wider community can stick their 2c/2p in on how useful it'd be and it can be prioritised appropriately.

**Reply by ajikur2901** [reply]
I have tried using a secondary header, but the secondary header is append in tbody, not in thread

![image](https://user-images.githubusercontent.com/48973903/228411702-52d435da-8b30-4e34-967e-0073469a72c8.png)

are there any options for a secondary header append in thread ?

**Comment by lrljoe** [comment]
I'm very confused as to what the issue here is to be honest?

If you use the setThAttributes() method to set the top header attributes, then you can certainly pass a colspan into that.
Then you can use the secondaryHeader as the main header, and pass any classes etc you want to using setSecondaryHeaderTdAttributes()

e.g.
![image](https://user-images.githubusercontent.com/104938042/228676740-b08ee37d-56c6-4e5c-bff7-4f68e7447180.png)

or
![image](https://user-images.githubusercontent.com/104938042/228677175-e96a86b6-d4c9-447e-8101-6650f80b8d81.png)

Keep in mind that the sortable functionality is not designed to work on secondary/tertiary etc headers.  The complexities involved there would be quite painful!
